website/project/model.py

```diff
@@ -157,19 +157,121 @@ class MetaData(GuidStoredObject):
     date_modified = fields.DateTimeField(auto_now=datetime.datetime.utcnow)
 
 
-def validate_comment_reports(value, *args, **kwargs):
+def validate_reports(value, *args, **kwargs):
     for key, val in value.iteritems():
         if not User.load(key):
             raise ValidationValueError('Keys must be user IDs')
         if not isinstance(val, dict):
             raise ValidationTypeError('Values must be dictionaries')
-        if 'category' not in val or 'text' not in val:
+        if (
+            'category' not in val or
+            'text' not in val or
+            'date' not in val or
+            'retracted' not in val
+        ):
             raise ValidationValueError(
-                'Values must include `category` and `text` keys'
+                ('Values must include `date`, `category`, ',
+                 '`text`, `retracted` keys')
             )
 
 
-class Comment(GuidStoredObject):
+class SpamMixin(StoredObject):
```

Comment:
In order to slim down `model.py`, let's move this mixin to its own module in `website.project.spam`.